File: Nian_Karl/QL_Nian/src/ql/ast/expression/binary/Plus.java
package ql.ast.expression.binary;

import ql.ast.expression.Binary;
import ql.ast.expression.Expression;
import ql.ast.value.Int;

public class Plus extends Binary {
	
	private Expression left;
	private Expression right;
	
	public Plus(Expression left, Expression right) {
		super(left, right);
	}

	@Override
	public Int evaluate() {
		Integer l = (Integer) left.evaluate().getValue();
		Integer r = (Integer) right.evaluate().getValue();
		return new Int(l + r);
	}
	
}

File: fabien/src/QL/operators/Plus.py

from NumericOperand import NumericOperand

class Plus(NumericOperand):
    def evaluate(self, answers):
        return self.left.evaluate(answers) + \
               self.right.evaluate(answers)

    def __repr__(self):
        return "%s + %s" % (self.left, self.right)
